hibernate-core/src/main/java/org/hibernate/EntityFilterException.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate;
+
+import java.util.Locale;
+
+import org.hibernate.annotations.FilterDef;
+
+import jakarta.persistence.EntityNotFoundException;
+
+/**
+ * Exception thrown if a filter would make a to-one association {@code null},
+ * which could lead to data loss.
+ * Even though filters are applied to load-by-key operations,
+ * a to-one association should never of an entity should never refer to an entity that is filtered.
+ *
+ * @see FilterDef#applyToLoadByKey()
+ */
+public class EntityFilterException extends EntityNotFoundException {
```

Comment:
Not sure about the name of this guy. Can we come up with something more descriptive?

Reply:
We talked about it here but could not come up with anyting better.  Make a suggestion rather than just "I don't like this one" ;)

Reply:
Haha, obviously I don't have a better suggestion of I would have :)

Reply:
How about `FilterIntegrityException`?

Reply:
I prefer the current one over that personally